resources/type-system/inference.md

```diff
@@ -680,6 +686,38 @@ The constraint solution for a type variable `X` with respect to a constraint set
 Note that the constraint solution is a type schema, and hence may contain
 occurences of the unknown type.
 
+#### Constraint solution for a set of type variables
+
+The constraint solution for a set of type variables `{X0, ..., Xn}` with respect
+to a constraint set `C`, with previous solution `{T0, ..., Tn}`, is defined to
+be the set of type schemas `{U0, ..., Un}` such that:
+  - If `Ti` is known (that is, does not contain `_`), then `Ui = Ti`.  _(Note
+    that the upcoming "variance" feature will relax this rule so that it only
+    applies to type variables without an explicitly declared variance.)_
+  - Otherwise, let `Vi` be the constraint solution for the type variable `Xi`
+    with respect to the constraint set `C`.
+  - If `Vi` is not known (that is, it contains `_`), then `Ui = Vi`.
+  - Otherwise, if `Xi` does not have an explicit bound, then `Ui = Vi`.
+  - Otherwise, let `Bi` be the bound of `Xi`.  Then, let `Bi'` be the type
```

Comment:
"be the type" -> "be the type schema" I think.  You are substituting in type schemas for type variables, so the result is a type schema.

Reply:
Fixed, thanks